source/js/modules/three/common/get-raw-shader-material-attrs.js

```diff
@@ -28,14 +28,31 @@ export default (uniforms) => (
 
       uniform sampler2D map;
 
+      struct optionsStruct {
+        float hueShift;
+      };
+
+      uniform optionsStruct options;
+
       varying vec2 vUv;
 
-      void main() {
+      vec3 hueShift(vec3 color, float hue) {
+        const vec3 k = vec3(0.57735, 0.57735, 0.57735);
```

Comment:
В glsl можно сокращать создание векторов с одинаковыми значениями для всех элементов:
```suggestion
        const vec3 k = vec3(0.57735);
```